codalab/model/bundle_model.py

```diff
@@ -588,17 +588,36 @@ def set_waiting_for_worker_startup_bundle(self, bundle, job_handle):
             }
             self.update_bundle(bundle, bundle_update, connection)
 
-    def set_starting_bundle(self, bundle, user_id, worker_id):
+    def set_starting_bundle(self, bundle, worker, request_cpus, request_gpus):
         """
         Sets the bundle to STARTING, updating the last_updated metadata. Adds
         a worker_run row that tracks which worker will run the bundle.
+
+        Return a dict consisting of the updated worker_run_row if success or None otherwise
         """
+        user_id = worker['user_id']
+        worker_id = worker['worker_id']
+
+        unused_cpuset = set(worker['cpuset'])
```

Comment:
One thing that we should make sure we support is that there's going to be a PR that has a worker backed by AWS batch which effectively has an infinite number of CPUs/GPUs, in which case cpuset/gpuset will be something like None.